master/buildbot/changes/gerritchangesource.py

```diff
@@ -29,6 +30,33 @@
 from buildbot.util import httpclientservice
 
 
+def canonicalize_event(event):
+    """
+    Return an event dictionary which is consistent between the gerrit
+    event stream and the gerrit event log formats.
+    """
+    # For "patchset-created" the events-log JSON looks like:
+    #   "project": {"name": "buildbot"}
+    # while the stream-events JSON looks like:
+    #   "project": "buildbot"
+    # so we canonicalize them to the latter
+    if ("change" in event and "project" in event["change"] and
+            isinstance(event["change"]["project"], dict) and
+            "name" in event["change"]["project"]):
+        event = dict(event)
+        event["change"] = dict(event["change"])
+        event["change"]["project"] = event["change"]["project"]["name"]
+    return event
```

Comment:
you probably want to add an empty line before the `return` statement

Reply:
Done